### ISSUE TITLE
YOLOv8 Object Detection Model Training: Copy_Paste Augmentation Options Not Being Applied

### ISSUE THREAD

**Issue by leeeunhyeok1** [ultralytics/ultralytics#22449]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, I’m training my Yolov8 (nano) object detection model with 48 training images and 5 validation images. 

This is my training command, and copy_paste option is used
<img width="927" height="107" alt="Image" src="https://github.com/user-attachments/assets/5e7ea23e-10ba-4456-8d02-3aa233be0402" />

These are my train_batch 0, 1, 2.jpg files, and here copy_paste options is not applied. However, other options like translate, hsv are properly applied. 

![Image](https://github.com/user-attachments/assets/1d19ad2d-9e40-4bdf-9ff1-7a8923c32570)
![Image](https://github.com/user-attachments/assets/23b561f5-eb1d-490a-ad1a-7d21600a64b5)
![Image](https://github.com/user-attachments/assets/1b21323d-d6a8-457c-934e-5e62c774618d)

I installed Ultralytics using pip install ultralytics. 

I have no idea what the problem is.. please help..!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @leeeunhyeok1, thanks for opening this issue and for using Ultralytics 🚀 This is an automated response — an Ultralytics engineer will follow up soon to assist you further.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report (e.g., copy_paste augmentation not appearing in your train_batch samples), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. To speed things up, please include:
-Exact training command you ran (full CLI) and any YAML configs used
-Your `ultralytics` version and environment details (OS, Python, PyTorch/CUDA)
-A very small sample of your dataset (a few images+labels) that reproduces the issue
-The saved `train_batch*.jpg` previews you referenced
-Any relevant logs showing augmentation settings applied at start-up

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Sharing these details will help us quickly pinpoint what’s happening with copy_paste in your setup 🧩

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
`copy_paste` is only for segmentation training:
https://docs.ultralytics.com/modes/train/#augmentation-settings-and-hyperparameters

**Comment by leeeunhyeok1** [comment]
thanks!

**Comment by glenn-jocher** [comment]
Glad it’s resolved—if you want augmentation for object detection, tune `mosaic`, `mixup`, and `cutmix` instead; for example:
```bash
yolo detect train data=data.yaml model=yolo11n.pt imgsz=640 epochs=100 mosaic=1.0 mixup=0.2 cutmix=0.2 close_mosaic=10
```
Details and tips are in the Ultralytics [data augmentation guide](https://docs.ultralytics.com/guides/yolo-data-augmentation/).
